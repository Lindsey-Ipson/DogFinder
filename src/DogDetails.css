.DogDetails {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  margin-top: 50px;
}

.DogDetails img {
  width: 300px;
  border-radius: 10px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);

}