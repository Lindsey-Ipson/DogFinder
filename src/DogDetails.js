import React from 'react';
import "./DogDetails.css";

function DogDetails ({ dog }) {
  console.log('fvcytfcyu')
  console.log('DOG HERE:', dog)
  console.log('NAME HERE:', dog.name)
  console.log('AGE HERE:', dog.age)
  console.log('FACTS HERE:', dog.facts)
  console.log('SRC HERE:', dog.src)

  const factLis = dog.facts.map(fact => <li>{fact}</li>)

  return (
    <div className="DogDetails">
      <h1>Meet {dog.name}!</h1>
      <img src={dog.src} alt={dog.name} />
      <ul>
        <li>{dog.name} is {dog.age} years old.</li>
        {factLis}
       </ul>
    </div>
  )
};

export default DogDetails;