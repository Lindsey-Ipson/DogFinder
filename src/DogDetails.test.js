import React from 'react';
import { useParams } from 'react-router-dom';
import { render } from '@testing-library/react';
import DogDetails from './DogDetails';

const mockDog = {
  name: 'Example Dog',
  age: 3,
  src: 'example.jpg',
  facts: ['Fact 1', 'Fact 2', 'Fact 3'],
};

test('renders DogDetails component with correct content', () => {
  const { getByText, getByAltText } = render(<DogDetails dog={mockDog} />);

  // Check if the dog name is rendered
  expect(getByText(`Meet ${mockDog.name}!`)).toBeInTheDocument();

  // Check if the image is rendered with the correct alt text
  expect(getByAltText(mockDog.name)).toBeInTheDocument();

  // Check if the age and name are rendered
  expect(getByText(`${mockDog.name} is ${mockDog.age} years old.`)).toBeInTheDocument();

  // Check if each fact is rendered
  mockDog.facts.forEach((fact) => {
    expect(getByText(fact)).toBeInTheDocument();
  });
});

test('matches snapshot', () => {
  const { asFragment } = render(<DogDetails dog={mockDog} />);
  expect(asFragment()).toMatchSnapshot();
});

