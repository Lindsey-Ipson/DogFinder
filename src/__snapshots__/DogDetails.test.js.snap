// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DogDetails"
  >
    <h1>
      Meet Example Dog!
    </h1>
    <img
      alt="Example Dog"
      src="example.jpg"
    />
    <ul>
      <li>
        Example Dog is 3 years old.
      </li>
      <li>
        Fact 1
      </li>
      <li>
        Fact 2
      </li>
      <li>
        Fact 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
